packages/block-editor/src/components/provider/index.js

```diff
@@ -1,170 +1,30 @@
-		if ( this.unsubscribe ) {
-			this.unsubscribe();
-		}
-	}
-
-	/**
-	 * Given a registry object, overrides the default dispatch behavior for the
-	 * `core/block-editor` store to interpret a state change and decide whether
-	 * we should call `onChange` or `onInput` depending on whether the change
-	 * is persistent or not.
-	 *
-	 * This needs to be done synchronously after state changes (instead of using
-	 * `componentDidUpdate`) in order to avoid batching these changes.
-	 *
-	 * @param {WPDataRegistry} registry Registry from which block editor
-	 *                                  dispatch is to be overridden.
-	 */
-	attachChangeObserver( registry ) {
-		if ( this.unsubscribe ) {
-			this.unsubscribe();
-		}
+function BlockEditorProvider( props ) {
+	const { children, settings } = props;
 
-		const {
-			getBlocks,
-			getSelectionStart,
-			getSelectionEnd,
-			isLastBlockChangePersistent,
-			__unstableIsLastBlockChangeIgnored,
-		} = registry.select( 'core/block-editor' );
+	const { updateSettings } = useDispatch( 'core/block-editor' );
+	useEffect( () => {
+		updateSettings( settings );
+	}, [ settings ] );
 
-		let blocks = getBlocks();
-		let isPersistent = isLastBlockChangePersistent();
+	// Syncs the entity provider with changes in the block-editor store.
+	useBlockSync( props );
 
-		this.unsubscribe = registry.subscribe( () => {
-			const { onChange = noop, onInput = noop } = this.props;
-
-			const newBlocks = getBlocks();
-			const newIsPersistent = isLastBlockChangePersistent();
-
-			if (
-				newBlocks !== blocks &&
-				( this.isSyncingIncomingValue ||
-					__unstableIsLastBlockChangeIgnored() )
-			) {
-				this.isSyncingIncomingValue = null;
-				blocks = newBlocks;
-				isPersistent = newIsPersistent;
-				return;
-			}
-
-			if (
-				newBlocks !== blocks ||
-				// This happens when a previous input is explicitely marked as persistent.
-				( newIsPersistent && ! isPersistent )
-			) {
-				// When knowing the blocks value is changing, assign instance
-				// value to skip reset in subsequent `componentDidUpdate`.
-				if ( newBlocks !== blocks ) {
-					this.isSyncingOutcomingValue.push( newBlocks );
-				}
-
-				blocks = newBlocks;
-				isPersistent = newIsPersistent;
-
-				const selectionStart = getSelectionStart();
-				const selectionEnd = getSelectionEnd();
-
-				if ( isPersistent ) {
-					onChange( blocks, { selectionStart, selectionEnd } );
-				} else {
-					onInput( blocks, { selectionStart, selectionEnd } );
-				}
-			}
-		} );
-	}
-
-	render() {
-		const { children } = this.props;
-
-		return children;
-	}
+	return children;
 }
 
-export default compose( [
-	withRegistryProvider,
-	withDispatch( ( dispatch ) => {
-		const { updateSettings, resetBlocks, resetSelection } = dispatch(
-			'core/block-editor'
-		);
-
-		return {
-			updateSettings,
-			resetBlocks,
-			resetSelection,
-		};
-	} ),
-] )( BlockEditorProvider );
+export default compose( [ withRegistryProvider ] )( BlockEditorProvider );
```

Comment:
compose is useless here I guess

Reply:
yep. could probably export `withRegistryProvider` as a hook now too since it is already a functional component

Reply:
Eh, probably not worth the effort. It still needs to wrap the component